python/lsst/cp/pipe/ptc/cpSolvePtcTask.py

```diff
@@ -219,8 +219,14 @@ def run(self, inputCovariances, camera=None, detId=0):
                 means, variances, and exposure times
                 (`lsst.ip.isr.PhotonTransferCurveDataset`).
         """
+        # Find the ampNames from a non-dummy ptc.
```

Comment:
In `cpPtcExtract`, the dummy datasets had the ampNames: https://github.com/lsst/cp_pipe/blob/main/python/lsst/cp/pipe/ptc/cpExtractPtcTask.py#L314 
So are you actually building a pipeline that does not have `cpPtcExtract` as a previous step of `cpPTcSolve`?

Reply:
The original code was relying on all input PTC datasets having the same unique set of `ampNames`.  The bug in `ip_isr` allowed each instance of the `PhotonTransferCurveDataset` to update the global list of `ampNames` for all instances, as it updated the class definition instead of just the instance definition.  This meant that if someone was lazy about constructed the additional padding datasets (me), then the `Solve` task would attempt to iterate over the union of all `ampNames` found in all datasets.  Fixing the bug ensured that the inputs only knew about their own `ampNames`, so the `np.unique` no longer was needed to prune the duplicates from the first input, but it did require that the first input have the same set of `ampNames` as all other inputs.  Adding the requirement the `ampNames` come from a not-DUMMY dataset allows the additional padding datasets to be lazy, and makes the content of the DUMMY datasets unimportant; they only serve to make the middleware accounting correct, and have no science value.